### PR TITLE
use `towncrier` to handle change log entries and update new pull request checklist

### PR DESCRIPTION
**Pull request by zacharyburnett**
follow-on to https://github.com/spacetelescope/romancal/pull/1375

# old checklist

**Checklist**
- [ ] Schema changes discussed at RAD Review Board meeting
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [ ] Updated relevant roman_datamodels utilities and tests
- [ ] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/XXX/

# new checklist

## Tasks
- [ ] update or add relevant `rad` tests
- [x] update relevant docstrings and / or `docs/` page
- [ ] Does this PR change schema?
  - [ ] schema changes were discussed at RAD Review Board meeting
- [ ] Does this PR change any API used downstream? (if not, label with `no-changelog-entry-needed`)
  - [x] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [ ] update relevant `roman_datamodels` utilities and tests
  - [ ] [start a `romancal` regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) with this branch installed (`"git+https://github.com/<fork>/rad@<branch>"`)

<details><summary>news fragment change types...</summary>

- ``changes/<PR#>.apichange.rst``: change to public API
- ``changes/<PR#>.bugfix.rst``: fixes an issue
- ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
</details

### CODE REVIEW

**Comment by PaulHuwe** [comment]
I thought the existing changelog file would be converted to JSON, which would then be used going forward. Did I misunderstand?

**Comment by zacharyburnett** [comment]
> I thought the existing changelog file would be converted to JSON, which would then be used going forward. Did I misunderstand?

I don't know about that, I considered modifying the existing changelog entries to make the issue numbers have links to the GitHub issue but I'm not sure how useful old changelog entries are (or who looks at them). This would just change how new sections are added to `CHANGES.rst` upon release

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -8,8 +8,22 @@ Closes #
...
 This PR addresses ...
 
-**Checklist**
-- [ ] Schema changes discussed at RAD Review Board meeting
-- [ ] Added entry in `CHANGES.rst` under the corresponding subsection
-- [ ] Updated relevant roman_datamodels utilities and tests
-- [ ] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/XXX/
+<!-- if you can't perform these tasks due to permissions, please ask a maintainer to do them -->
+## Tasks
+- [ ] update or add relevant `rad` tests
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:13]:
```suggestion
- [ ] Update or add relevant `rad` tests.
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+- [ ] update relevant docstrings and / or `docs/` page
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:14]:
```suggestion
- [ ] Update relevant docstrings and / or `docs/` page.
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+- [ ] Does this PR change schema?
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:15]:
```suggestion
- [ ] Does this PR change any schema files?
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+  - [ ] schema changes were discussed at RAD Review Board meeting
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:16]:
```suggestion
  - [ ] Schema changes were discussed at RAD Review Board meeting.
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+- [ ] Does this PR change any API used downstream? (if not, label with `no-changelog-entry-needed`)
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:17]:
```suggestion
- [ ] Does this PR change any API used downstream? (If not, label with `no-changelog-entry-needed`.)
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:18]:
```suggestion
  - [ ] Write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types).
```

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+  - [ ] [start a `romancal` regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) with this branch installed (`"git+https://github.com/<fork>/rad@<branch>"`)
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:19]:
```suggestion
  - [ ] Start a `romancal` regression test (https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) with this branch installed (`"git+https://github.com/<fork>/rad@<branch>"`).
```

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+  - [ ] update relevant `roman_datamodels` utilities and tests
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:20]:
```suggestion
  - [ ] Update relevant `roman_datamodels` utilities and tests.
```

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -8,8 +8,22 @@ Closes #
...
+
+<details><summary>news fragment change types...</summary>
```
Comment (PaulHuwe) [inline .github/pull_request_template.md:22]:
```suggestion
<details><summary>News fragment change types:</summary>
```

**Review by PaulHuwe — CHANGES_REQUESTED** [review]
Just some formatting changes for future PRs.

**Review by PaulHuwe — APPROVED** [review]
LGTM

**Pull request merged**